src/Plugin/dgi_migrate_alter/spreadsheet/DssiNodeAlter.php

```diff
@@ -0,0 +1,225 @@
+ *   migration_id = "dssi_node"
+ * )
+ */
+class DssiNodeAlter extends MigrationAlterBase implements MigrationAlterInterface {
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \JsonException
+   */
+  public function alter(array &$migration): void {
+    $process =& $migration['process'];
+
+    $process['field_resource_type'][1]['bundle'] = 'library_of_congress_resource_typ';
+
+    // Add Text (plain) delimited fields.
+    $plain_text_delimited_fields_to_add = [
+      'field_local_contexts' => ['local_contexts', ';'],
+      'field_access_id' => ['access_id', ';'],
+    ];
+
+    foreach ($plain_text_delimited_fields_to_add as $field => $field_info) {
+      [$source, $delimiter] = $field_info;
+      $process[$field] = [
+        [
+          'plugin' => 'skip_on_empty',
+          'source' => $source,
+          'method' => 'process',
+        ],
+        [
+          'plugin' => 'dgi_migrate.process.explode',
+          'delimiter' => $delimiter,
+        ],
+        [
+          'plugin' => 'skip_on_empty',
+          'method' => 'row',
+          'message' => "No usable values in $field[1].",
+        ],
+      ];
+    }
+
+    // Add Text (plain) non-delimited fields.
+    $plain_text_non_delimited_fields_to_add = [
+      'field_scale' => 'scale',
+    ];
+
+    foreach ($plain_text_non_delimited_fields_to_add as $field => $source) {
+      $process[$field] = [
+        [
+          'plugin' => 'skip_on_empty',
+          'source' => $source,
+          'method' => 'process',
+        ],
+      ];
+    }
+
+    // Add tertiary delimiters.
+    $tetiary_delimiter_fields_to_add = [
+      ['field_origin_information', 'field_place', 'place', '^'],
+      ['field_origin_information', 'field_publisher', 'publisher', '^'],
+      ['field_origin_information', 'field_edition', 'edition', '^'],
+      ['field_origin_information', 'field_issuance', 'issuance', '^'],
+      ['field_origin_information', 'field_frequency', 'frequency', '^'],
+    ];
+    $expected_child_fields = [
+      'field_event_type', 'field_place', 'field_date_created',
+      'field_date_issued', 'field_date_captured', 'field_date_valid', 'field_date_modified',
+      'field_other_date', 'field_copyright_date', 'field_publisher', 'field_edition',
+      'field_issuance', 'field_frequency',
+    ];
+
+    foreach ($tetiary_delimiter_fields_to_add as $field) {
+      $parent_field = $field[0];
+      $child_field = $field[1];
+      $source_key = $field[2];
+      $delimiter = $field[3];
+
+      // Ensure that the keys in 'values' are a subset of $allowed_child_fields.
+      $values_keys = array_keys($process[$parent_field][2]['values']);
+      $is_subset = !array_diff($values_keys, $expected_child_fields);
+      \assert(
+        $is_subset,
+        'Keys in $process['
+        . $parent_field . '][2][\'values\'] are not a subset of allowed child fields: '
+        . \json_encode($values_keys, JSON_THROW_ON_ERROR)
+      );
```

Comment:
Checking that `$process[$parent_field][2]['plugin']` is whatever paragraph generation plugin is probably sufficient. Could possibly check the bundle; though, that might not be useful. Like: Imagine that the upstream migration (or some other alteration) adds or removes another plugin in the flow _before_ our offset `2`. Maybe, they legitimately handle moving to another bundle (and all the repercussions of doing so, including moving the fields over), might be fine.

On the other hand, if a step is added or removed earlier in the processing of the given field, then it would introduce an offset that breaks this, as `2` would no longer be correct.